tracing-subscriber/src/registry/sharded.rs

```diff
@@ -97,9 +108,33 @@ impl Registry {
     fn get(&self, id: &Id) -> Option<Guard<'_, DataInner>> {
         self.spans.get(id_to_idx(id))
     }
+
+    // `start_close` creates a guard which tracks how many layers have
+    // processed a close event via the `CLOSE_COUNT` thread-local. Once
+    // the `CLOSE_COUNT` is 0, the registry knows that is is safe to
+    // remove a span. It does so via the Drop implementation on
+    // `CloseGuard`.
```

Comment:
I guess this answers the "how does it work" question in the comment I left on the other doc comment?

Reply:
Yep, I did some more writing.